File: Fragment/app/src/main/java/com/rikkei/training/fragment/MediaManager.java
package com.rikkei.training.fragment;

import android.content.Context;
import android.media.MediaPlayer;

public class MediaManager {
    private int resId;
    private Context context;
    private MediaPlayer player;

    public MediaManager(int resId, Context context) {
        this.resId = resId;
        this.context = context;
    }

    public void create() {
        player = MediaPlayer.create(context, resId);
    }

    public void start() {
        if (player != null) {
            player.start();
        }
    }

    public void pause() {
        if (player != null) {
            player.pause();
        }
    }

    public void stop() {
        if (player != null) {
            player.stop();
        }
    }

    public void seekTo(int position){
        if (player!= null){
            player.seekTo(position);
        }
    }
    public int getCurrentPosition(){
        if (player!= null){
            return player.getCurrentPosition();
        }
        return 0;
    }
}

File: FragmentKotlin/app/src/main/java/com/rikkei/training/fragmentkotlin/MediaManager.kt
package com.rikkei.training.fragmentkotlin

import android.content.Context
import android.media.MediaPlayer

class MediaManager {
    private var resId: Int = 0
    private var mContext: Context? = null
    private var mPlayer: MediaPlayer? = null

    constructor(resId: Int, mContext: Context?) {
        this.resId = resId
        this.mContext = mContext
    }

    fun create() {
        mPlayer = MediaPlayer.create(mContext,resId)
    }

    fun start() {
        if (mPlayer != null) {
            mPlayer!!.start()
        }
    }

    fun pause() {
        if (mPlayer != null) {
            mPlayer!!.pause()
        }

    }

    fun stop() {
        if (mPlayer != null) {
            mPlayer!!.stop()
        }

    }

    fun seekTo(position: Int) {
        if (mPlayer != null) {
            mPlayer!!.seekTo(position)
        }
    }

    fun getCurrentPossition(): Int {
        if (mPlayer != null) {
            return mPlayer!!.currentPosition
        }
        return 0
    }
}